mode/markdown/markdown.js

```diff
@@ -51,24 +51,28 @@ CodeMirror.defineMode("markdown", function(cmCfg, modeCfg) {
   if (modeCfg.strikethrough === undefined)
     modeCfg.strikethrough = false;
 
+  // Allow class names to be overridden by user-provided classes.
+  if (modeCfg.classOverrides === undefined)
+    modeCfg.classOverrides = {};
+
   var codeDepth = 0;
 
-  var header   = 'header'
-  ,   code     = 'comment'
-  ,   quote    = 'quote'
-  ,   list1    = 'variable-2'
-  ,   list2    = 'variable-3'
-  ,   list3    = 'keyword'
-  ,   hr       = 'hr'
-  ,   image    = 'tag'
-  ,   formatting = 'formatting'
-  ,   linkinline = 'link'
-  ,   linkemail = 'link'
-  ,   linktext = 'link'
-  ,   linkhref = 'string'
-  ,   em       = 'em'
-  ,   strong   = 'strong'
-  ,   strikethrough = 'strikethrough';
+  var header   = modeCfg.classOverrides.header || "header"
+  ,   code     = modeCfg.classOverrides.code || "comment"
+  ,   quote    = modeCfg.classOverrides.quote || "quote"
+  ,   list1    = modeCfg.classOverrides.list1 || "variable-2"
+  ,   list2    = modeCfg.classOverrides.list2 || "variable-3"
+  ,   list3    = modeCfg.classOverrides.list3 || "keyword"
+  ,   hr       = modeCfg.classOverrides.hr || "hr"
+  ,   image    = modeCfg.classOverrides.image || "tag"
+  ,   formatting = modeCfg.classOverrides.formatting || "formatting"
+  ,   linkinline = modeCfg.classOverrides.linkinline || "link"
+  ,   linkemail = modeCfg.classOverrides.linkemail || "link"
+  ,   linktext = modeCfg.classOverrides.linktext || "link"
+  ,   linkhref = modeCfg.classOverrides.linkhref || "string"
+  ,   em       = modeCfg.classOverrides.em || "em"
+  ,   strong   = modeCfg.classOverrides.strong || "strong"
+  ,   strikethrough = modeCfg.classOverrides.strikethrough || "strikethrough";
```

Comment:
How about moving the styles into an object, say `styles`, and using a for/in loop to do the overriding? That's less noisy and easier to extend in the future (it does make the patch bigger, requiring you to rename all references to these, but that should be safe and improve clarity -- I've found myself wondering what, say, `code` refers to in a given context).

Also, since these aren't direct CSS classes (they will be prefixed with `cm-`), we tend to call them styles or token types, not classes. It'd be good if the option name reflected that.